src/librustc/mir/interpret/allocation.rs

```diff
@@ -16,15 +16,20 @@ use std::borrow::Cow;
 #[derive(Clone, Debug, Eq, PartialEq, PartialOrd, Ord, Hash, RustcEncodable, RustcDecodable)]
 pub struct Allocation<Tag=(),Extra=()> {
     /// The actual bytes of the allocation.
-    /// Note that the bytes of a pointer represent the offset of the pointer
-    pub bytes: Vec<u8>,
+    /// Note that the bytes of a pointer represent the offset of the pointer. Bytes of undefined
+    /// allocations may be allocated lazily as bytes are initialized but either none or all bytes
+    /// are allocated.
+    crate bytes: Vec<u8>,
```

Comment:
Can't we restrict this further? Preferrably these would be private to this module only.